src/cluster/k8sClientHandler.go

```diff
@@ -512,11 +512,15 @@ func GetSecrets(k8sClient *kubernetes.Clientset, label string, namespace string,
 
 	secrets, err := k8sClient.CoreV1().Secrets(namespace).Get(context.Background(), name, metav1.GetOptions{})
 
-	if err != nil {
+	if err != nil && !strings.Contains(err.Error(), "not found") {
```

Comment:
Yeah this is not a good practice.

Reply:
if there is an error that means we don't have data in secret, right? And since we are only printing the error we shouldn't be bothered with the different cases of error that might get generated. Or are we going to handle different types of errors?

Reply:
I see the reasoning here. We need to fix the rbac rules for discovery-engine then. But if you want to handle that separately then that's also fine ig.
@nyrahul Wdyt?

Reply:
Okay, we can fix that in a separate PR. Otherwise LGTM